commit 2dad0bde7d8f8d6f451154c5f7e8ef454ba056fc
Author:     Yan Cheng <58191769+yanchengnv@users.noreply.github.com>
AuthorDate: Fri Nov 17 15:37:36 2023 -0500
Commit:     GitHub <noreply@github.com>
CommitDate: Fri Nov 17 15:37:36 2023 -0500

    Support sys vars for job config and support parameterized template in job config (#2145)
    
    * support sys vars for job config
    
    * address pr reviews
    
    * support os env vars used in job config
    
    * support os env vars used in job config
    
    ---------
    
    Co-authored-by: Chester Chen <512707+chesterxgchen@users.noreply.github.com>
---
 nvflare/private/fed/client/client_engine.py   | 9 +++++++--
 nvflare/private/fed/client/client_executor.py | 6 ++++++
 2 files changed, 13 insertions(+), 2 deletions(-)

diff --git a/nvflare/private/fed/client/client_engine.py b/nvflare/private/fed/client/client_engine.py
index 7d622b40..4e197897 100644
--- a/nvflare/private/fed/client/client_engine.py
+++ b/nvflare/private/fed/client/client_engine.py
@@ -21,7 +21,7 @@ import threading
 
 from nvflare.apis.event_type import EventType
 from nvflare.apis.fl_component import FLComponent
-from nvflare.apis.fl_constant import MachineStatus, SystemComponents, WorkspaceConstants
+from nvflare.apis.fl_constant import FLContextKey, MachineStatus, SystemComponents, WorkspaceConstants
 from nvflare.apis.fl_context import FLContext, FLContextManager
 from nvflare.apis.workspace import Workspace
 from nvflare.fuel.utils.network_utils import get_open_ports
@@ -74,6 +74,9 @@ class ClientEngine(ClientEngineInternalSpec):
             private_stickers={
                 SystemComponents.DEFAULT_APP_DEPLOYER: AppDeployer(),
                 SystemComponents.JOB_META_VALIDATOR: JobMetaValidator(),
+                SystemComponents.FED_CLIENT: client,
+                FLContextKey.SECURE_MODE: self.client.secure_train,
+                FLContextKey.WORKSPACE_ROOT: args.workspace,
             },
         )
 
@@ -149,6 +152,7 @@ class ClientEngine(ClientEngineInternalSpec):
 
         open_port = get_open_ports(1)[0]
 
+        server_config = list(self.client.servers.values())[0]
         self.client_executor.start_app(
             self.client,
             job_id,
@@ -158,7 +162,8 @@ class ClientEngine(ClientEngineInternalSpec):
             allocated_resource,
             token,
             resource_manager,
-            list(self.client.servers.values())[0]["target"],
+            target=server_config["target"],
+            scheme=server_config.get("scheme", "grpc"),
         )
 
         return "Start the client app..."
diff --git a/nvflare/private/fed/client/client_executor.py b/nvflare/private/fed/client/client_executor.py
index 3727fb38..9ced2ba8 100644
--- a/nvflare/private/fed/client/client_executor.py
+++ b/nvflare/private/fed/client/client_executor.py
@@ -46,6 +46,7 @@ class ClientExecutor(ABC):
         token,
         resource_manager,
         target: str,
+        scheme: str,
     ):
         """Starts the client app.
 
@@ -59,6 +60,7 @@ class ClientExecutor(ABC):
             token: token from resource manager
             resource_manager: resource manager
             target: SP target location
+            scheme: SP target connection scheme
         """
         pass
 
@@ -147,6 +149,7 @@ class ProcessExecutor(ClientExecutor):
         token,
         resource_manager: ResourceManagerSpec,
         target: str,
+        scheme: str,
     ):
         """Starts the app.
 
@@ -160,6 +163,7 @@ class ProcessExecutor(ClientExecutor):
             token: token from resource manager
             resource_manager: resource manager
             target: SP target location
+            scheme: SP connection scheme
         """
         new_env = os.environ.copy()
         if app_custom_folder != "":
@@ -185,6 +189,8 @@ class ProcessExecutor(ClientExecutor):
             + str(client.cell.get_internal_listener_url())
             + " -g "
             + target
+            + " -scheme "
+            + scheme
             + " -s fed_client.json "
             " --set" + command_options + " print_conf=True"
         )
